pkg/serverhandler/handlertest/test_service_project.go

```diff
@@ -210,3 +211,21 @@ func TestServiceProject_UpsertApplication(t *testing.T, factory Factory) {
 		}
 	})
 }
+
+func TestServiceProject_InvalidName(t *testing.T, factory Factory) {
+	ctx := context.Background()
+	require := require.New(t)
+	client, _ := factory(t)
+
+	// GRPC Gateway interprets ../ as a path traversal, and therefore we cannot allow
+	// '../' in any fields we use as path tokens.
+	project := ptypes.TestProject(t, &pb.Project{
+		Name: "../../",
```

Comment:
We should probably also enforce this at the state level when we go to upsert a project name, if we don't already have some kind of name validation there? 🤔

Reply:
The main thing is going forward, any new API endpoints now have to know about this path validation. If it's in the state package, every current and future API endpoint will behave this way no matter how many new ones we add.